### PR TITLE
Support preferring new packages over installed packages.

### PR DESCRIPTION
**Pull request by johntyree**
This adds a flag to the policy that allows us to turn off the preference for packages that are already installed on the system. It can be toggled at any time, but the envisioned use case is to select it at initialization and use it throughout.

Additionally a bug is fixed wherein the same version of a package available in the installed repo is chosen from the remote repo. The test scenario provided suffered from this issue prior to this PR.

### CODE REVIEW

File: simplesat/dependency_solver.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 class DependencySolver(object):
     def __init__(self, pool, remote_repositories, installed_repository,
-                 policy=None, use_pruning=True):
+                 policy=None, use_pruning=True, prefer_installed=True):
```
Comment (cournape) [inline simplesat/dependency_solver.py]: I would prefer putting that choice in the policy class, to avoid polluting the signature of `DependencySolver`.

Comment (johntyree) [inline simplesat/dependency_solver.py]: It's just getting passed through here. I'll refactor.

**Comment by cournape** [comment]
Besides the arg comment, I would suggest to add another scenario similar to the `update_reverse_dependencies_no_prefer_installed.yaml` one, but using the prefered installed first policy, to easily compare the solution. Or do we already have such an example ?

**Comment by johntyree** [comment]
`update_reverse_dependencies.yaml` is the same scenario, but with the original solution using the installed first policy.

File: simplesat/dependency_solver.py (hunk 1 above)
Comment (johntyree) [inline simplesat/dependency_solver.py]: :+1:

**Comment by johntyree** [comment]
@cournape I have removed the new kwarg from the dep solver API and added some comments to the two scenario files to indicate how they correspond.

**Comment by cournape** [comment]
LGTM

**Comment by johntyree** [comment]
Somehow this PR is not merged into master.

**Comment by johntyree** [comment]
Scratch that, some of my local refs were stale and reporting no updates somehow. Everything's probably fine here.

**Pull request merged**